pkgs/development/compilers/futhark/package.nix

```diff
@@ -0,0 +1,49 @@
+{ mkDerivation, alex, array, base, bifunctors, binary, blaze-html
+, bytestring, containers, data-binary-ieee754, directory
+, directory-tree, dlist, extra, file-embed, filepath, gitrev, happy
+, haskeline, hpack, HUnit, json, language-c-quote, mainland-pretty
+, markdown, mtl, neat-interpolation, parallel, parsec, process
+, process-extras, QuickCheck, random, raw-strings-qq, regex-tdfa
+, srcloc, stdenv, template-haskell, temporary, test-framework
+, test-framework-hunit, test-framework-quickcheck2, text
+, th-lift-instances, transformers, vector, vector-binary-instances
+, zlib
+}:
+mkDerivation {
+  pname = "futhark";
+  version = "0.6.2";
+  src = /nix/store/5n2kpq21b3hm0yfr7wsinh5v24y81s98-source;
```

Comment:
While a bit ugly, this is always overridden by the `default.nix`.

Reply:
Why not take the `src` as an argument? This is too ugly.

Reply:
This is autogenerated by cabal2nix, I can't just change what it generates. I also don't want to have to manually edit the file every time an update comes out. It would be possible to use `cabal2nix`'s git fetching. But that has the problem that it doesn't use the github tarballs, it fetches all submodules, and futhark has a super big submodule that takes forever to fetch. This may get better with https://github.com/NixOS/cabal2nix/issues/260. I don't know of a cleaner solution than this.